.github/workflows/unittests.yml

```diff
@@ -39,8 +39,36 @@ jobs:
           python -m pip install --user -e .[extras]
       - name: Test project
         run: |
-          python -m pytest --cov=./ --cov-report=xml --durations=50 tests/
+          python -m pytest --cov=./ --cov-report=xml --device="cpu" --durations=50 tests/
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1.0.10
         with:
           env_vars: OS,PYTHON
+
+  unittest-gpu:
+    runs-on: ubuntu-latest
+    strategy: 
+      fail-fast: false
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install Linux dependencies
+        run: sudo apt-get install libopenblas-dev
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```

Comment:
I find this "Secrets are not passed to workflows that are triggered by a pull request from a fork. [Learn more](https://docs.github.com/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets). "

Reply:
Some useful information. [Empty-env-secrets](https://stackoverflow.com/questions/58737785/github-actions-empty-env-secrets), [pull_request_target](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target)

Reply:
I didn't find any mention of pull request in the linked doc. The closest sentence I found was "With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository." This seems to mean that the workflows in forks don't have access to the same credential, which is expected.

Reply:
The pull_request_target seems to suggest that this PR needs to be merged first in order to trigger the action
```
pull_request_target
This event is similar to pull_request, except that it runs in the context of the base repository of the pull request, rather than in the merge commit. This means that you can more safely make your secrets available to the workflows triggered by the pull request, because only workflows defined in the commit on the base repository are run. For example, this event allows you to create workflows that label and comment on pull requests, based on the contents of the event payload.
```

Reply:
Sorry for the misleading. The sentence "Secrets are not passed to workflows that are triggered by a pull request from a fork" is from Settings>Secrets>The second sentence for repos. The linked docs are related issues about env secrets.

Reply:
I think to use the pull_request_target, this line needs to be changed: https://github.com/dmlc/gluon-nlp/blob/749acece178e2137ed341e0ebe8437782263339c/.github/workflows/unittests.yml#L3